testsuite/testcases/src/quorum_store_onchain_enable_test.rs

```diff
@@ -0,0 +1,131 @@
+            )
+            .await
+        });
+
+        std::thread::sleep(duration / 2);
+
+        runtime.block_on(async {
+
+            let root_cli_index = cli.add_account_with_address_to_cli(
+                swarm.chain_info().root_account().private_key().clone(),
+                swarm.chain_info().root_account().address(),
+            );
+
+            let current_consensus_config: OnChainConsensusConfig = bcs::from_bytes(
+                &rest_client
+                    .get_account_resource_bcs::<Vec<u8>>(
+                        CORE_CODE_ADDRESS,
+                        "0x1::consensus_config::ConsensusConfig",
+                    )
+                    .await
+                    .unwrap()
+                    .into_inner(),
+            )
+            .unwrap();
+
+            let inner = match current_consensus_config {
+                OnChainConsensusConfig::V1(inner) => inner,
+                OnChainConsensusConfig::V2(_) => panic!("Unexpected V2 config"),
+            };
+
+            // Change to V2
+            let new_consensus_config = OnChainConsensusConfig::V2(ConsensusConfigV1 { ..inner });
+
+            let update_consensus_config_script = format!(
+                r#"
+        script {{
+            use aptos_framework::aptos_governance;
+            use aptos_framework::consensus_config;
+            fun main(core_resources: &signer) {{
+                let framework_signer = aptos_governance::get_signer_testnet_only(core_resources, @0000000000000000000000000000000000000000000000000000000000000001);
+                let config_bytes = {};
+                consensus_config::set(&framework_signer, config_bytes);
+            }}
+        }}
+        "#,
+                generate_blob(&bcs::to_bytes(&new_consensus_config).unwrap())
+            );
+
+            cli.run_script_with_default_framework(root_cli_index, &update_consensus_config_script)
+                .await
+        })?;
+
+        std::thread::sleep(duration / 2);
+
+        // Wait for all nodes to synchronize and stabilize.
+        info!("Waiting for the validators to be synchronized.");
+        runtime.block_on(async {
+            swarm
+                .wait_for_all_nodes_to_catchup(Duration::from_secs(MAX_NODE_LAG_SECS))
+                .await
+        })?;
+
+        Ok(())
+    }
+}
+
+impl NetworkTest for QuorumStoreOnChainEnableTest {
+    fn run<'t>(&self, ctx: &mut aptos_forge::NetworkContext<'t>) -> anyhow::Result<()> {
+        <dyn NetworkLoadTest>::run(self, ctx)
+    }
+}
+
+fn generate_blob(data: &[u8]) -> String {
+    let mut buf = String::new();
+
+    write!(buf, "vector[").unwrap();
+    for (i, b) in data.iter().enumerate() {
+        if i % 20 == 0 {
+            if i > 0 {
+                writeln!(buf).unwrap();
+            }
+        } else {
+            write!(buf, " ").unwrap();
+        }
+        write!(buf, "{}u8,", b).unwrap();
+    }
+    write!(buf, "]").unwrap();
+    buf
```

Comment:
When generating a blob or something like this, it may be helpful to add a comment that describes what it'll look like.